src/managers/KeybindManager.hpp

```diff
@@ -12,7 +12,7 @@ class CPluginSystem;
 
 struct SKeybind {
     std::string key          = "";
-    int         keycode      = -1;
+    uint32_t    keycode      = 0;
```

Comment:
why this change? keycode 0 is valid, iirc. Maybe some `std::optional`s here?

Reply:
>why this change?

Explained in commit message here [`6fcc7a9` (#4304)](https://github.com/hyprwm/Hyprland/pull/4304/commits/6fcc7a9179e20d7e57f02e50068ca1206b4b25c1)

Isn't keycode 0 used for lack-of-keycode in xkb too?

Reply:
k then